qiskit/assembler/assemble_circuits.py

```diff
@@ -116,8 +116,7 @@ def assemble_circuits(circuits, run_config, qobj_id, qobj_header):
                                                        mask="0x%X" % mask,
                                                        relation='==',
                                                        val="0x%X" % val,
-                                                       register=conditional_reg_idx,
-                                                       validate=False)
+                                                       register=conditional_reg_idx)
```

Comment:
We don't want to do it because it's exceedingly slow. @kdk made this change recently in #3370 with some numbers (it previously wasn't an even an option). In my testing of a worst case `qiskit.compiler.assemble()` call for a qobj of 100x a 53x2048 circuit showed that this PR without validation took ~2min and~7min with validation. (vs ~9min for marshmallow with validation). This PR makes validation opt-in throughout the stack so we don't need to set this anymore. It also makes validate only an option on the top level Qobj clasess instead of each individual field class so this results in an error setting validate to false here.